### PR TITLE
docs/integration: recommend using stable paths in fstab and system.conf

### PR DESCRIPTION
**Pull request by jluebbe**
Fixes: #1149

### CODE REVIEW

**Review by obbardc — APPROVED** [review]

File: docs/integration.rst [hunk 1]
```diff
@@ -55,6 +55,24 @@ partition for U-Boot), :ref:`data partition(s) <sec-data-storage>` or similar.
 Since changing the partition layout is hard or even impossible to change in the
 field, make sure it meets both current and possible future requirements.
 
+.. note::
+
+  The ``/etc/fstab`` in your system's root FS (and RAUC's ``system.conf``)
+  should normally use stable paths to refer to the partitions or devices.
```
Comment (ejoerns) [inline docs/integration.rst:61]:
```suggestion
  should normally use stable paths to refer to partitions or devices.
```

File: docs/integration.rst [hunk 2]
```diff
@@ -55,6 +55,24 @@ partition for U-Boot), :ref:`data partition(s) <sec-data-storage>` or similar.
...
+  Especially a filesystem UUIDs (``UUID=<uuid>`` or
```
Comment (ejoerns) [inline docs/integration.rst:62]:
```suggestion
  Especially filesystem UUIDs (``UUID=<uuid>`` or
```

File: docs/integration.rst [hunk 3]
```diff
@@ -55,6 +55,24 @@ partition for U-Boot), :ref:`data partition(s) <sec-data-storage>` or similar.
...
+  ``/dev/disk/by-uuid/<uuid>``) should *not* be used, as they are likely to be
+  different after an update.
+
+  Depending on your system design and firmware, good stable paths can be:
+
+  * plain device names: ``/dev/sda``, ``/dev/mmcblk0p1``, ``/dev/nvme0n1p1``, ``/dev/mapper/…``
+  * topology-based symlinks: ``/dev/disk/by-path/…`` (preferable if available)
+  * partition-table-uuid-based symlinks: ``/dev/disk/by-partuuid/…`` (breaks if
+    using the same disk image on i.e. eMMC & SD)
```
Comment (ejoerns) [inline docs/integration.rst:71]:
```suggestion
  * partition-table-UUID-based symlinks: ``/dev/disk/by-partuuid/…`` (breaks if
    using the same disk image on e.g. both eMMC & SD card)
```

File: docs/integration.rst [hunk 4]
```diff
@@ -55,6 +55,24 @@ partition for U-Boot), :ref:`data partition(s) <sec-data-storage>` or similar.
...
+
+  The ``/etc/fstab`` in your system's root FS (and RAUC's ``system.conf``)
+  should normally use stable paths to refer to the partitions or devices.
+  Especially a filesystem UUIDs (``UUID=<uuid>`` or
+  ``/dev/disk/by-uuid/<uuid>``) should *not* be used, as they are likely to be
+  different after an update.
+
+  Depending on your system design and firmware, good stable paths can be:
+
+  * plain device names: ``/dev/sda``, ``/dev/mmcblk0p1``, ``/dev/nvme0n1p1``, ``/dev/mapper/…``
```
Comment (ejoerns) [inline docs/integration.rst:68]: Should we mention that these may depend on storage enumeration order in some cases?

**Review by ejoerns — CHANGES_REQUESTED** [review]

**Review by ejoerns — APPROVED** [review]

**Pull request merged**